src/arns.lua

```diff
@@ -361,73 +370,59 @@ function arns.getTokenCost(intendedAction)
 	local baseFee = demand.baseFeeForNameLength(#name)
 	local intent = intendedAction.intent
 	local qty = tonumber(intendedAction.quantity)
+	local record = arns.getRecord(name)
+	local currentTimestamp = tonumber(intendedAction.currentTimestamp)
 
 	assert(type(intent) == "string", "Intent is required and must be a string.")
 	assert(type(name) == "string", "Name is required and must be a string.")
 	if intent == "Buy-Record" then
-		assert(purchaseType == "lease" or purchaseType == "permabuy", "PurchaseType is invalid.")
-		if purchaseType == "lease" then
-			assert(years >= 1 and years <= 5, "Years is invalid. Must be an integer between 1 and 5")
-		end
+		-- stub the process id as it is not required for this intent
+		local processId = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+		arns.assertValidBuyRecord(name, years, purchaseType, processId)
```

Comment:
this part seems odd to be in the non-test code. without digging into the function I wonder why wouldn't `assertValidBuyRecord` take an optional processID instead

Reply:
I guess you're just getting the cost here, whereas the assertion function may be used in later on in the creation flow